File: HackerRank/Algorithms/Strings/Beautiful_binary_string.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */
    int length;
    string s;
    int step_count = 0;
    cin >> length >> s;
    for (int i = 0; i < length;) {
        if (s.substr(i,3) == "010") {
            step_count++;
            i = i + 3;
        }
        else {
            ++i;
        }
    }
    cout << step_count << endl;
    return 0;
}

File: HackerRank/Algorithms/Strings/Beautiful_binary_string.java
import java.util.Scanner;

public class Solution {

    public static void main(String[] args) {
        /* Enter your code here.
           Read input from STDIN.
           Print output to STDOUT.
           Your class should be named Solution. */
        Scanner scan = new Scanner(System.in);
        int length = Integer.parseInt(scan.nextLine());
        String str = scan.nextLine();
        scan.close();
        int stepCount = 0;
        for(int i = 0; i < length-2;) {
            if(str.substring(i, i+3).equals("010")) {
                stepCount++;
                i = i + 3;
            }
            else {
                ++i;
            }
        }
        System.out.println(stepCount);
    }
}
